### PR TITLE
Fix/lighter xpress interface

### PR DESCRIPTION
**Pull request by pet-mit**
### Change 1
As the official OR-Tools repository before v9.2 used the Make build system, the generated JNI was named "main_research_linear_solver".
Since we use CMake instead, in order to be compatible with their jars published on maven central, we had to rename the JNI generated by CMake "main_research_linear_solver" as well.

However, since [v9.3](https://github.com/google/or-tools/releases/tag/v9.3), Google also use the CMake build system. So now their JNI is called "main" (see [original file](https://github.com/google/or-tools/blob/stable/ortools/linear_solver/java/CMakeLists.txt)).
So now we can go back to naming it "main" like Google, to be compatible with their jars.

This change reverts [this commit](https://github.com/rte-france/or-tools/commit/56000cc36b830e7155a7ff6dee7fad15946e27a9).

### Change 2
Newer XPRESS versions can remove some functions from the API. As long as these are not used in the XPRESS interface, this should not bother us.
However, currently the XPRESS interface imports **all** functions that were available in the XPRESS API v8.13. Thus it is no longer compatible with newer versions (e.g. v9.0).
This change removes all functions that are imported but not used in the interface, allowing it to be compatible with new versions (at least more easily adaptable).

### CODE REVIEW

**Review by sgatto — APPROVED** [review]
Looks good to me, but I am not able to judge all the changes of the xpress interface

**Comment by pet-mit** [comment]
Java build tested with XPRESS 9.0, both issues are resolved

**Review by flomnes — CHANGES_REQUESTED** [review]
See file **ortools/xpress/parse_header_xpress.py**.

It has a set of functions that should be skipped. I suggest changing the logic a little bit to
- Keep only the required #defines
- Keep only the required functions

Then run it to re-generate environment.h/.cpp

**Comment by pet-mit** [comment]
> See file **ortools/xpress/parse_header_xpress.py**.
> 
> It has a set of functions that should be skipped. I suggest changing the logic a little bit to
> 
> * Keep only the required #defines
> * Keep only the required functions
> 
> Then run it to re-generate environment.h/.cpp

good advice. I did not know this script existed

File: ortools/xpress/parse_header_xpress.py [hunk 1]
```diff
@@ -138,21 +258,35 @@ def parse(self, filepath):
...
-        print('------------------- define -------------------')
+        print('------------------- define (to copy in the define part of environment.cc) -------------------')
         print(self.__define)
 
-        print('------------------- assign -------------------')
+        print('------------------- assign (to copy in the assign part of environment.cc) -------------------')
         print(self.__assign)
 
+    def print_missing_elements(self):
+        if len(self.__missing_required_defines) > 0:
```
Comment (flomnes) [inline ortools/xpress/parse_header_xpress.py:271]:
```suggestion
        if self.__missing_required_defines:
```

File: ortools/xpress/parse_header_xpress.py [hunk 2]
```diff
@@ -138,21 +258,35 @@ def parse(self, filepath):
...
+            print('------WARNING------ missing required defines -------------------')
+            for symbol in self.__missing_required_defines:
+                print(symbol)
+
+        if len(self.__missing_required_functions) > 0:
```
Comment (flomnes) [inline ortools/xpress/parse_header_xpress.py:276]: same

File: ortools/xpress/parse_header_xpress.py [hunk 3]
```diff
@@ -138,21 +258,35 @@ def parse(self, filepath):
...
+            print('------WARNING------ missing required functions -------------------')
+            for name in self.__missing_required_functions:
+                print(name)
+
+        if len(self.__missing_required_defines) > 0 or len(self.__missing_required_functions) > 0:
```
Comment (flomnes) [inline ortools/xpress/parse_header_xpress.py:281]: same

File: ortools/xpress/parse_header_xpress.py [hunk 4]
```diff
@@ -138,21 +258,35 @@ def parse(self, filepath):
...
 
-        print('------------------- assign -------------------')
+        print('------------------- assign (to copy in the assign part of environment.cc) -------------------')
         print(self.__assign)
 
+    def print_missing_elements(self):
+        if len(self.__missing_required_defines) > 0:
+            print('------WARNING------ missing required defines -------------------')
+            for symbol in self.__missing_required_defines:
+                print(symbol)
```
Comment (flomnes) [inline ortools/xpress/parse_header_xpress.py:274]:
```suggestion
            print(self.__missing_required_defines)
```

File: ortools/xpress/parse_header_xpress.py [hunk 5]
```diff
@@ -36,23 +37,142 @@ def __init__(self):
...
+    def should_fun_be_imported(self, name):
+        return name in self.__required_functions
 
     def write_define(self, symbol, value):
+        if not self.should_define_be_imported(symbol):
+            print('skipping ' + symbol)
+            return
+
         self.__header += f'#define {symbol} {value}\n'
+        self.__missing_required_defines.remove(symbol)
```
Comment (flomnes) [inline ortools/xpress/parse_header_xpress.py:164]: 1. Remove `return` statement, use `if..else`. 
2. Remove `not`, invert content of `if` and `else` blocks.

**Review by flomnes — CHANGES_REQUESTED** [review]

**Review by flomnes — APPROVED** [review]

**Pull request merged**